### ISSUE TITLE
Would you be open to rewriting this in typescript?

### ISSUE THREAD

**Issue by PeterKottas** [rcdexta/react-trello#285]
I've seen that you mentioned you were considering rewriting this lib to typescript (https://github.com/rcdexta/react-trello/issues/115), I have recently done this for a much bigger lib (https://github.com/react-page/react-page/pull/629) and it worked out rly well. What you have here is pretty small so I could do it in no time. It would also help my use case if there were reliable (autogenerated) typescript definitions. The only hurdle I see is all PRs should be merged before I begin. What do you think?

**Comment by dapi** [comment]
@PeterKottas hi!

Would you like to migrate to typescript or just add types?

**Comment by dapi** [comment]
I'm not sure about migrating to the typescript whole project. But it is will be nice to have types for a data structure (lane, card). Is it possible?

**Comment by PeterKottas** [comment]
I would prefer to migrate the project in its entirety to ts. It's not a lot of code, I could be done with it by the end of today. The build won't change much as babel can do typescript these days. The thing is, you can never guarantee the reliability of typescript typings if it's not autogenerated. The only way to do it is having code in ts. Also, the coding experience is very nice. That's IMHO of course.

**Comment by PeterKottas** [comment]
Also this way, you won't have to keep an eye on keeping typescript typings up to date. Less maintanence is always good.

**Comment by gavinbelson** [comment]
I would suggest to keep it as is. There are hidden rabbit holes with migrations

**Comment by PeterKottas** [comment]
Sorry to hear that but obviously, I respect the decision. Depending on how much we'll be using this lib, chances are I'll fork, rewrite and continue the development there. Will keep you posted. Cheers.

**Comment by mattsrobot** [comment]
@PeterKottas any update on this? I'm interested in Typescript support for react-trello.

**Comment by phoenixcoded20** [comment]
Me too.

**Comment by matheuscamarques** [comment]
I need this
